src/System.Device.Gpio/System/Device/Pwm/Channels/PwmChannel.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Device.Pwm
+{
+    /// <summary>
+    /// Represents a single PWM channel.
+    /// </summary>
+    public abstract partial class PwmChannel : IDisposable
+    {
+        /// <summary>
+        /// The frequency in hertz.
+        /// </summary>
+        public abstract double Frequency { get; }
+
+        /// <summary>
+        /// The duty cycle percentage represented as a value between 0.0 and 1.0.
+        /// </summary>
+        public abstract double DutyCyclePercentage { get; set; }
+
+        /// <summary>
+        /// Starts the PWM channel.
+        /// </summary>
+        public abstract void Start();
+
+        /// <summary>
+        /// Stops the PWM channel.
+        /// </summary>
+        public abstract void Stop();
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
not needed since you don't have a finalizer

Reply:
Presumably the type is unsealed (otherwise it wouldn't need the pattern at all), so the SuppressFinalize here doesn't hurt... and does catch if a derived type adds one.

Reply:
perhaps we should ask a language team to inject it in every Dispose team then

Reply:
They don't like injecting things :smile:. But the pattern is

```C#
public void Dispose()
{
    Dispose(true);
    GC.SuppressFinalize(this);
}

protected virtual void Dispose(bool disposing)
{
    ...
}
```